### ISSUE TITLE
train.py的训练结果无法直接应用到eval.py中

### ISSUE THREAD

**Issue by XhivaW** [hzy46/fast-neural-style-tensorflow#1]
谢谢分享的项目！
使用train.py训练后能在对应路径下看到检查点及最终的训练数据如附图。
训练方法按照说明中一样进行 vgg16已放入对应文件夹 train2014在同一主目录下 基于wave.yml修改名称和对应图片路径得到新的配置文件
但是其中的fast-style-model.ckpt-done.data-00000-of-00001无法被eval.py读取。
报错如下：
\--------------------------------
DataLossError (see above for traceback): Unable to open table file /home/tensorflow/neural_style/fast-neural-style-tensorflow-master/trained_model/sketch.ckpt-done: Data loss: not an sstable (bad magic number): perhaps your file is in a different file format and you need to use a different restore operator?
\--------------------------------
看了看代码似乎新的模型导入只要使用默认结构就不用特别声明吧？还是train.py的输出还需处理才能直接使用？又或者是我训练步骤有问题？
![train_result](https://cloud.githubusercontent.com/assets/12463092/21445309/56595642-c8f2-11e6-847f-bcdfa009bbef.png)

**Comment by hzy46** [comment]
1. eval的时候不需要读vgg16和train2014。所以肯定不是它们的问题。
2. 你说无法读取fast-style-model.ckpt-done.data-00000-of-00001。但是你这里报错的时候读的是 /home/tensorflow/neural_style/fast-neural-style-tensorflow-master/trained_model/sketch.ckpt-done。这个是不是有点问题？

**Comment by XhivaW** [comment]
谢谢回复

1.我的意思是训练过程是严格按照说明中的 并且运行过程中没问题
2.这个只是我复制fast-style-model.ckpt-done.data-00000-of-00001到trained_model文件夹中并改名为sketch.ckpt-done 东西还是一样的 eval.py代码内部的引用我也确定过没问题

主要想问一下您提供下载的训练好的模型是像我上面这样直接使用train.py得到的done.data这个还是另外处理过了？

**Comment by hzy46** [comment]
没有经过额外处理。另外我的版本是Tensorflow 0.11，你参考一下。

**Comment by XhivaW** [comment]
谢谢回复
我也是0.11的。。 
再确认一下 上面train.py生成的检查点以及对应模型是正常的吗？
其他我自己再研究研究好啦
再次谢谢分享的项目

**Comment by hzy46** [comment]
是的呢。我并没有使用Saver的任何高级特性，就是简单的保存和恢复呀。

**Comment by XhivaW** [comment]
您好
我周末重新跑了一次 发现训练过程中最后会有如图错误 不知道是不是导致读取错误的原因 想问一下您在运行过程中有没有遇到过类似的问题
![bug](https://cloud.githubusercontent.com/assets/12463092/21473607/71c001a4-cb4c-11e6-8860-7c599ca4fcc5.png)

**Comment by XhivaW** [comment]
您好
我周末重新跑了一次 发现训练过程中最后会有如图错误 不知道是不是导致读取错误的原因 想问一下您在运行过程中有没有遇到过类似的问题
![bug3](https://cloud.githubusercontent.com/assets/12463092/21473794/08126b98-cb51-11e6-8bd9-46826c2cdd65.png)
文字如下
\-------------------------------------------
W tensorflow/core/framework/op_kernel.cc:975] Out of range: FIFOQueue '_0_input_producer' is closed and has insufficient elements (requested 1, current size 0)
	 [[Node: ReaderRead = ReaderRead[_class=["loc:@WholeFileReader", "loc:@input_producer"], _device="/job:localhost/replica:0/task:0/cpu:0"](WholeFileReader, input_producer)]]
\-------------------------------------------
39800次后 上述一样的内容连续出现5次 然后到达迭代次数上限自动终止


另外我试了试前面checkpoint如36000等保存的数据 同样无法读取。。。

**Comment by hzy46** [comment]
这不是错误。。只是达到两个epoch了。

**Comment by XhivaW** [comment]
问题已解决
TensorFlow 0.11中的某次更新增加了一种saver的存储格式(V2) 并设为新的默认格式了 您的代码是基于原有的V1格式写的
 
如果您希望保留V1格式的话 可以在[train.py](https://github.com/hzy46/fast-neural-style-tensorflow/blob/master/train.py)中对saver做如下修改

```python
saver = tf.train.Saver(variables_to_restore,write_version=tf.train.SaverDef.V1)
```
则代码将生成V1格式的保存结果 与您提供下载的已训练模型参数的格式一致

如果您希望使用V2格式的话 可以在储存最终结果时将checkpoints/fast-style-model.ckpt-done.data-00000-of-00001/fast-style-model.ckpt-done.index/fast-style-model.ckpt-done.meta 这四个文件单独放在同一目录下 运行[eval.py](https://github.com/hzy46/fast-neural-style-tensorflow/blob/master/eval.py)时 --model_file 指向存放这四个文件的目录即可 效果与指向V1格式的.data文件一致

**Comment by hzy46** [comment]
谢谢解决这个版本问题！
